### PR TITLE
Update to latest eslint tools

### LINKED ISSUE #15: Upgrade to eslint 2.0
eslint 2.0 is in release-candidate status, but once it's released we should adopt it.  We'll need to wait for eslint-config-zeal to support it first (see https://github.com/CodingZeal/eslint-config-zeal/issues/3).

### PR DESCRIPTION
**Pull request by randycoulman**
Upgrade to eslint 2.1, eslint-config-zeal 0.3.0, eslint-plugin-react
3.16.1, eslint-loader 1.30.0, and babel-eslint 4.1.8.

Fixes #15.

This needs resolutions to the following bugs before it can be merged:

- [x] https://github.com/eslint/eslint/issues/5227
- [x] https://github.com/MoOx/eslint-loader/issues/79
- [x] https://github.com/CodingZeal/eslint-config-zeal/issues/5

### CODE REVIEW

**Comment by randycoulman** [comment]
@lexun @seanculver @julianrogmans Final review please?

File: client/store/index.js [hunk 1]
```diff
@@ -18,8 +18,10 @@ const store = createStore(reducer, undefined,
   )
 )
 
-if (module.hot) {
-  module.hot.accept('../reducers', () => {
+const hotLoader = module.hot
```
Comment (seanculver) [inline client/store/index.js]: You can put this back the way it ways if you upgrade the latest eslint-config-zeal

**Comment by seanculver** [comment]
:+1: Merge when done reverting index.js

**Comment by seanculver** [comment]
LGTM!!

**Pull request merged**